File: php/test/Behavioral/MementoTest.php
<?

require_once 'src/Behavioral/Memento/DvdDetails.php';

class Behavioral_MementoTest extends PHPUnit_Framework_TestCase
{
   public function test()
   {
		$dvdMementoCaretaker;  

		$stars = array("Guy Pearce");
		$dvdDetails = new DvdDetails("Memento", $stars, '1'); 

		$this->assertNotNull($dvdDetails,'dvddetails is not null test');
		$this->assertEquals($dvdDetails->formatDvdDetails(), 'DVD: Memento, starring: Guy Pearce, encoding region: 1', 'first formatDvdDetails call');

		$dvdMementoCaretaker = $dvdDetails->createDvdMemento();
		//$this->assertEquals( var_export($dvdMementoCaretaker,true), '','var export the memento');
		$this->assertNotNull($dvdMementoCaretaker, 'test we could create a memento');

		$dvdDetails->addStar("edskdzkvdfb");  
		#//oops - Cappuccino on the keyboard!
		$this->assertEquals($dvdDetails->formatDvdDetails(),'DVD: Memento, starring: Guy Pearce, edskdzkvdfb, encoding region: 1','Add weird named star test');


		$dvdDetails->setDvdMemento($dvdMementoCaretaker);
		#//back off changes
		$this->assertEquals($dvdDetails->formatDvdDetails(),'DVD: Memento, starring: Guy Pearce, encoding region: 1','Revert dvddetails back test');

	}
}

File: go/behavioral/memento/memento_test.go

package memento

import (
	"testing"
	"../../lib/asserts"
)


func TestMemento (t *testing.T) {

	stars := make([]string,0)
	stars = append(stars,"Guy Pearce")
	dvdDetails := NewDvdDetails("Memento", stars, "1")
	dvdMementoCaretaker := dvdDetails.CreateDvdMemento()

	asserts.Equals( t, "Testing original state", 
		"DVD: Memento, starring: Guy Pearce, encoding region: 1",
		dvdDetails.FormatDvdDetails())
       
   dvdDetails.AddStar("edskdzkvdfb")
   asserts.Equals( t, "Testing after adding star",
		"DVD: Memento, starring: Guy Pearce, edskdzkvdfb, encoding region: 1",
		dvdDetails.FormatDvdDetails() )
       
	asserts.Equals( t, "Memento state", 
		"DVD: Memento, starring: Guy Pearce, encoding region: 1",
		dvdMementoCaretaker.ShowMemento())

	//show the memento
	dvdDetails.SetDvdMemento(dvdMementoCaretaker)

	//back off changes
	asserts.Equals( t, "Showing state revised to reflect memento state", 
		"DVD: Memento, starring: Guy Pearce, encoding region: 1",
		dvdDetails.FormatDvdDetails())
}      
